ui/src/cloud/actions/demodata.ts

```diff
@@ -46,39 +57,77 @@ export const getDemoDataBuckets = () => async (
   }
 }
 
-export const getDemoDataBucketMembership = (bucketID: string) => async (
+export const getDemoDataBucketMembership = (bucket: DemoBucket) => async (
   dispatch,
   getState: GetState
 ) => {
+  const state = getState()
   const {
     me: {id: userID},
-  } = getState()
+  } = state
+  const {id: orgID} = getOrg(state)
 
   try {
-    await getDemoDataBucketMembershipAJAX(bucketID, userID)
-
+    await getDemoDataBucketMembershipAJAX(bucket.id, userID)
     dispatch(getBuckets())
-    // TODO: check for success and error appropriately
-    // TODO: instantiate dashboard template
+
+    const template = DemoDataTemplates[bucket.name]
+    if (template) {
+      await createDashboardFromTemplate(template, orgID)
+    } else {
+      throw new Error(
+        `Could not find template for demodata bucket ${bucket.name}`
+      )
+    }
+
+    // TODO: notify success and error appropriately
   } catch (error) {
     console.error(error)
   }
 }
 
-export const deleteDemoDataBucketMembership = (bucketID: string) => async (
+export const deleteDemoDataBucketMembership = (bucket: DemoBucket) => async (
   dispatch,
   getState: GetState
 ) => {
+  const state = getState()
   const {
     me: {id: userID},
-  } = getState()
+  } = state
+  const {id: orgID} = getOrg(state)
 
   try {
-    await deleteDemoDataBucketMembershipAJAX(bucketID, userID)
+    await deleteDemoDataBucketMembershipAJAX(bucket.id, userID)
     dispatch(getBuckets())
 
-    // TODO: check for success and error appropriately
-    // TODO: delete associated dashboard
+    const dashboardsResp = await api.getDashboards({query: {orgID}})
```

Comment:
why do we need to refresh the dashboards here? and why isn't it using one of the dashboard actions that updates the normalized state?